src/App.tsx

```diff
@@ -1,34 +1,124 @@
+/* eslint-disable no-console */
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+enum SortType {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setLoading] = useState(true);
+  const [cases, setCase] = useState('');
+  const [query, setQuery] = useState('');
+  const [button, setButton] = useState(false);
+  const [todo, setTodo] = useState(todos[0]);
```

Comment:
I suppose here should be a null as an initial value 
```suggestion
  const [todo, setTodo] = useState<Todo | null>(null);
```